recipes/lora_finetune_single_device.py

```diff
@@ -694,7 +698,14 @@ def train(self) -> None:
                     prof.step()
 
                 self.epochs_run += 1
+                start_save_checkpoint = time.time()
```

Comment:
nit: would use `time.perf_counter()` here unless there's a strong reason not to. If I understand the two APIs correctly it shouldn't make a huge difference for this application, and we use `time.perf_counter()` elsewhere in the recipe, so nice to be consistent. But do lmk if there's a specific reason we should be using `time.time()`